src/processors/universal.ts

```diff
@@ -370,6 +369,136 @@ class UniversalProcessor extends BaseOrderProcessor {
     return unlockTx;
   }
 
+  private unlockOrder(
+    orderId: string,
+    order: OrderData,
+    context: OrderProcessorContext
+  ) {
+    const giveChain = order.give.chainId;
+    // filling batch queue
+    let orderIds = this.unlockBatchesOrderIdMap.get(giveChain);
+    if (!orderIds) {
+      orderIds = new Set();
+    }
+    orderIds.add(orderId);
+    this.unlockBatchesOrderIdMap.set(giveChain, orderIds);
+    this.ordersDataMap.set(orderId, order);
+
+    if (this.unlockBatchesLockedStatus.get(giveChain) === true) {
+      this.context.logger.debug(
+        `batch unlock processing is locked for ${giveChain}`
+      );
+      return;
+    }
+
+    this.unlockBatchesLockedStatus.set(giveChain, true);
```

Comment:
Looks like a bug: you must acquire a mutex only if you are going to send batch. So, if there is enough orders to unlock, then you acquire a mutex and start unlocking. Here you erroneously acquire a mutex and leave it unreleased.